File: backend/app/schemas/notification.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime


class NotificationActor(BaseModel):
    id: str
    full_name: Optional[str]
    username: Optional[str]
    avatar_url: Optional[str]
    is_verified: bool

    model_config = {"from_attributes": True}


class NotificationResponse(BaseModel):
    id: UUID
    user_id: UUID
    actor_id: UUID
    type: str
    entity_type: str
    entity_id: UUID
    entity_user_id: Optional[UUID]
    content: Optional[str]
    extra_json: Optional[str]
    is_read: bool
    read_at: Optional[datetime]
    created_at: datetime
    actor: Optional[NotificationActor] = None

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    notifications: list[NotificationResponse]
    total: int
    unread_count: int
    has_more: bool


class NotificationMarkReadRequest(BaseModel):
    notification_ids: list[str] = Field(default_factory=list)


class NotificationCountResponse(BaseModel):
    unread_count: int

File: frontend/src/types/notification.ts
export interface NotificationActor {
  id: string;
  full_name: string | null;
  username: string | null;
  avatar_url: string | null;
  is_verified: boolean;
}

export interface Notification {
  id: string;
  user_id: string;
  actor_id: string;
  type: "like" | "comment" | "friend_request" | "friend_accept" | "follow" | "mention" | "share" | "message" | "birthday" | "live" | "system";
  entity_type: "post" | "comment" | "message" | "user" | "live" | "story";
  entity_id: string;
  entity_user_id: string | null;
  content: string | null;
  extra_json: string | null;
  is_read: boolean;
  read_at: string | null;
  created_at: string;
  actor: NotificationActor | null;
}

export interface NotificationListResponse {
  notifications: Notification[];
  total: number;
  unread_count: number;
  has_more: boolean;
}

export interface NotificationCountResponse {
  unread_count: number;
}

export interface NotificationMarkReadRequest {
  notification_ids: string[];
}
